### DISCUSSION TITLE
YOLOv8 Autoshape

### DISCUSSION THREAD

**Discussion by itamapeleg** [ultralytics/ultralytics#4538] [Q&A]
Please help me identify the counterpart of YOLOv5's Autoshape in YOLOv8? I'm working on a project and need to understand the equivalent feature or functionality in YOLOv8. Any insights or explanations would be greatly appreciated

**Comment by glenn-jocher** [comment]
@itamapeleg hello! In YOLOv8, the functionality similar to YOLOv5's Autoshape is built into the model's `predict` mode. When you use YOLOv8 to make predictions, the model automatically handles the pre-processing of images, including resizing and padding, to ensure they're in the correct format for inference. This means you can pass images of various resolutions directly to the model, and it will adapt accordingly, streamlining the process for you. Happy coding! 😊🚀
